hnet/handler.go

```diff
@@ -265,6 +265,39 @@ func handleUserRelationshipRemove(stream *common.IOStream, player *Player) error
 	return nil
 }
 
+func handleLeaderboardRequest(stream *common.IOStream, player *Player) error {
+	request := ReadLeaderboardRequest(stream)
+
+	if request == nil {
+		return fmt.Errorf("failed to read leaderboard request")
+	}
+
```

Comment:
I think the default status should be `NotSubmitted`, if I recall correctly. Unknown was for checksum mismatches.

Reply:
I am not too sure, better if you check first.

Reply:
`NeedsUpdate` takes priority over ranked status anyways, so this shouldn't really matter in the long run.

Reply:
It's just how osu! does it, so no clue.

Reply:
I think this should be fine for now, if it's not in db, then for all we know it's unsubmitted, and if checksums don't match, we can assume client's map is outdated